engine/internal/autoscaler/keda.go

```diff
@@ -0,0 +1,129 @@
+package autoscaler
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	kedaclient "github.com/kedacore/keda/v2/pkg/generated/clientset/versioned/typed/keda/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// NewKedaScaler creates a new KedaScaler.
+func NewKedaScaler(config KedaConfig) Autoscaler {
+	return &KedaScaler{config: config}
+}
+
+// KedaScaler is a scaler that uses KEDA to scale.
+type KedaScaler struct {
+	config KedaConfig
+
+	client   kedaclient.KedaV1alpha1Interface
+	recorder record.EventRecorder
+	logger   logr.Logger
+}
+
+// SetupWithManager sets up the multi-autoscaler with the Manager.
+func (s *KedaScaler) SetupWithManager(mgr ctrl.Manager) error {
+	s.logger = mgr.GetLogger().WithName("kedascaler")
+	s.recorder = mgr.GetEventRecorderFor("inference-scaler")
+
+	c, err := kedaclient.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		return fmt.Errorf("failed to create KEDA client: %s", err)
+	}
+	s.client = c
+
+	return mgr.Add(s)
+}
+
+// Start starts the multi-autoscaler.
+func (s *KedaScaler) Start(ctx context.Context) error {
+	s.logger.Info("Starting keda-scaler")
+	// TODO(aya): support KEDA external scaler
+	return nil
+}
+
+// NeedLeaderElection implements LeaderElectionRunnable and always returns true.
+func (s *KedaScaler) NeedLeaderElection() bool {
+	return true
```

Comment:
Just curious, why does this need to be `true`? No need to change, but just wondered there is no ham if two engines attempt to create the same ScaleObject (and one of them succeeds).

Reply:
Ah, yes, we don't need to enable this for the current scaler. I originally thought I would provide the inference metrics as a KEDA external scaler, which I'm working on now, but if we only send an activation request through the external scaler pusher, this may be unnecessary. I’ll turn off this for now.

Reply:
I see. Thanks!